pkg/image/controller/trigger/image_trigger_controller_test.go

```diff
@@ -936,16 +936,17 @@ func benchmark_1_imageStream(identity, maxTags, sequence int32, round, index int
 // updateBuildConfigImages updates the LastTriggeredImageID field on a build config.
 func updateBuildConfigImages(bc *buildapi.BuildConfig, tagRetriever trigger.TagRetriever) (*buildapi.BuildConfig, error) {
 	var updated *buildapi.BuildConfig
-	for i, t := range bc.Spec.Triggers {
-		p := t.ImageChange
+	bcCopy := bc.DeepCopy()
```

Comment:
The point of this method is to return "new object with no shared memory, or nil".  If bc was mutated, GetInputReference is broken (mutating, and must be fixed) or DeepCopy is broken (mutating, and must be fixed).  I don't see how this change helps

Reply:
The extra deep copies you added in d4859926e1725c6f685e38aeff2dc4acee216111 make no sense now either.

Reply:
I see you didn't add those deep copies - that makes this worse - we did a deep copy before (and weren't seeing these issue?)  And then we update to 1.8 and suddenly we have race conditions?

I'm 90% sure DeepCopy has a race - the other alternative is that the reported race isn't actually the last reader (but I don't see how that could be).

Reply:
We should probably try to test that a random update to a build config doesn't mutate the original.

Reply:
cc @sttts i will continue to debug this tomorrow

Reply:
also I ran that test 20 times in cycle and was not able to hit the race